src/AWS.Distro.OpenTelemetry.AutoInstrumentation/OtlpAwsSpanExporter/OtlpAwsSpanExporter.cs

```diff
@@ -87,9 +87,10 @@ public override ExportResult Export(in Batch<Activity> batch)
     {
         using IDisposable scope = SuppressInstrumentationScope.Begin();
 
-        byte[]? serializedSpans = OtlpExporterUtils.SerializeSpans(batch, this.processResource);
+        byte[] serializedData = new byte[750000];
+        int serializedDataLength = OtlpExporterUtils.WriteTraceData(ref serializedData, 0, this.processResource, batch);
 
-        if (serializedSpans == null)
+        if (serializedData == null || serializedDataLength == -1)
```

Comment:
Need to check buffer overflow of `serializedData`.

Reply:
There won't be any buffer overflow in this case because if the array size is small, the `WriteTraceData` will double it first and then write to it.